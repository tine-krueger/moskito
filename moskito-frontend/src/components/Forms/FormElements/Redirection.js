import { Redirect } from 'react-router-dom'
import styled from 'styled-components/macro'
import Delayed from '../../../hooks/useDelay'

export default function Redirection() {
    return (
        <RedParagraph data-testid='redirect'>Registrierung erfolgreich, Du wirst zum Login weitergeleitet.
            <Delayed waitBeforeShow={2500}>
                <Redirect to="/login"/>
            </Delayed>
        </RedParagraph>
    ) 
}

const RedParagraph = styled.p`
    color: var(--link);
`