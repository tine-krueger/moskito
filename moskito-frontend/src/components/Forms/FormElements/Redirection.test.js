import { render, screen, waitFor } from '@testing-library/react'
import 'jest-styled-components'
import Redirection from './Redirection'
import renderWithRouter from '../../../testSetup/setupTests'

jest.setTimeout(7000)

describe('Redirection', () => {

    it('renders correctly', () => {
        const { container } = render(<Redirection/>)
        expect(container).toMatchSnapshot()
    })

    it('redirects after 2.5 seconds', async() => {
        const { history } = renderWithRouter(<Redirection />)
        await waitFor(() => expect(history.location.pathname).toEqual('/login'), {timeout: 3000})
    })

    

})