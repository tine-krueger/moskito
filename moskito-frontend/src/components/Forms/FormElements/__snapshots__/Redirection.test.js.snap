// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirection renders correctly 1`] = `
.c0 {
  color: var(--link);
}

<div>
  <p
    class="c0"
    data-testid="redirect"
  >
    Registrierung erfolgreich, Du wirst zum Login weitergeleitet.
  </p>
</div>
`;
